src/flownet/ert/_create_ert_setup.py

```diff
@@ -47,28 +47,94 @@ def _create_observation_file(
         Nothing
 
     """
-    num_training_dates = round(len(schedule.get_dates()) * training_set_fraction)
+    num_dates = len(schedule.get_dates())
+    num_training_dates = round(num_dates * training_set_fraction)
 
     if yaml:
+        observations_yamlobs = obs_file
```

Comment:
When creating the three files you repeat a lot of code. You could consider using a for loop, maybe just a single one, that gets a list of settings you want to export an observation file for.

Like so (untested code, indentation is messy):
```python


export_settings = [ 
                             ["complete", 1, num_dates],
                             ["training", 1, num_training_dates],
                             ["test", num_training_dates + 1, num_training_dates + 1],
                             ["complete", 1, num_dates],
                             ["training", 1, num_training_dates],
                             ["test", num_training_dates + 1, num_dates],
                        ]

prefix_name = os.path.splitext(obs_file)

for _, setting in enumerate(export_settings):
    if yaml:
        obs_export_type = "yaml"
    else:
        obs_export_type = "ertobs"
    export_filename = f"{prefix_name}_{setting[0]}.{obs_export_type}"
    template = _TEMPLATE_ENVIRONMENT.get_template(f"observations.{obs_export_type}.jinja2")
   
    with open(export_filename , "w") as fh:
            fh.write(
                template.render(
                    {
                        "schedule": schedule,
                        "error_config": config.flownet.data_source.simulation.vectors,
                        "num_beginning_date": setting[1],
                        "num_end_date": setting[2]
                    }
                )
            )
```

This potentially replaces lines 53-137. It would however require that there is only 1 template for ertobs, and 1 template for yaml obs. Which is directly another point I'd like to raise: by increasing the number of template the update job for a developer becomes even worse compared to updating two files, as we have to do per today. If there is no good argument for have 6 templates for 2 different files, I'd suggest to drop 4 of them.